media/CircleEditor/view-sidebar.js

```diff
@@ -254,77 +267,217 @@ sidebar.NodeSidebar = class {
+    constructor(host, node, isCustom) {
+        this._host = host;
+        this._node = node;
+        this._elements = [];
+        this._attributes = [];
+        this._isCustom = isCustom;
+        this._attributesBox = host.document.createElement('div');
+
+        this._editObject = {
+            name: 'custom',
+            _attribute: {},
+            _nodeIdx: parseInt(this._node.location),
+            _subgraphIdx: this._node._subgraphIdx
+        };
+
+        const sortedAttributes = node.attributes.slice();
+        sortedAttributes.sort((a, b) => {
+            const au = a.name.toUpperCase();
+            const bu = b.name.toUpperCase();
+            return (au < bu) ? -1 : (au > bu) ? 1 : 0;
+        });
+        this._addHeader('Attributes');
+        let index = 0;
+        for (const attribute of sortedAttributes) {
+            this._addAttribute(attribute.name, attribute, index);
+            index++;
+        }
+        if (isCustom === true) {
+            const addAttribute = this._host.document.createElement('div');
+            addAttribute.className = 'sidebar-view-item-value-add';
+            addAttribute.innerText = '+ New Attributes';
+            addAttribute.addEventListener('click', () => {
+                this.add();
+            });
+            this._attributesBox.appendChild(addAttribute);
+            this._elements.push(this._attributesBox);
+        }
+    }
+
+    _addHeader(title) {
+        const headerElement = this._host.document.createElement('div');
+        headerElement.className = 'sidebar-view-header';
+        headerElement.innerText = title;
+        this._elements.push(headerElement);
+    }
+
+    _addAttribute(name, attribute, index) {
+        const item =
+            new NodeAttributeView(this._host, attribute, this._isCustom, index, this._node);
         item.on('show-graph', (sender, graph) => {
             this._raise('show-graph', graph);
         });
-        const view = new sidebar.NameValueView(this._host, name, item);
+        const view = new sidebar.NameValueView(this._host, name, item, index, 'attribute');
         this._attributes.push(view);
-        this._elements.push(view.render());
+        if (this._isCustom === true) {
+            this._attributesBox.appendChild(view.render());
+        } else {
+            this._elements.push(view.render());
+        }
+    }
+
+    add() {
+        this._editObject._attribute.name = this._node.type.name;
+        const keys = [];
+        for (const key of this._node.attributes) {
+            keys.push(key.name);
+            this._editObject._attribute[key.name] = key.value;
+            this._editObject._attribute[key.name + '_type'] = key.type;
+        }
+        keys.push('attribute');
+        this._editObject._attribute['attribute'] = 'value';
+        this._editObject._attribute['attribute_type'] = 'string';
+        this._editObject._attribute.keys = keys;
+
+        // vscode.postMessage({
```

Comment:
If this will be enabled, please leave TODO comment

```suggestion
        // TODO Enable posting message
        // vscode.postMessage({
```